examples/quickstarts/helidon-quickstart-se/src/main/java/io/helidon/examples/quickstart/se/GreetService.java

```diff
@@ -100,6 +106,26 @@ private void sendResponse(ServerResponse response, String name) {
         response.send(returnObject);
     }
 
+    private static <T> T processErrors(Throwable ex, ServerRequest request, ServerResponse response) {
+         if (ex.getCause() instanceof JsonParsingException) {
+
+            LOGGER.log(Level.FINE, "Error parsing JSON body", ex);
+            JsonObject jsonErrorObject = JSON.createObjectBuilder()
+                .add("error", "Error parsing JSON body")
+                .build();
+            response.status(Http.Status.INTERNAL_SERVER_ERROR_500).send(jsonErrorObject);
+        } else if (ex.getCause() instanceof JsonException){
+
+            LOGGER.log(Level.FINE, "Empty request body", ex);
```

Comment:
@romain-grecourt Thanks for the good input. Sending a JSON Array throws a `ClassCastException`. If you send a JSON object with multiple keys, objects or arrays instead of strings, a `JsonParsingException` is thrown.

Right now, I'm still sending the same parsing message for these since the exception details are either `Unexpected char ...` or `Invalid token=...`. I can work on adding more custom messages depending on the parsing error if it's more appropriate.

Reply:
I'm sending minimally more informative messages for `Invalid token` errors vs `Unexpected char` errors.

`JsonParsingException` cases:

* Invalid token: JSON object is fine but the JSON schema wasn't expected
* Unexpected char if JSON is completely invalid

Are these messages appropriate?

Reply:
I don't think this we should specialize the error messages that much. Instead I'd say you should simply handle the two kinds of error that should return a 400 or a 500 and use a generic error message.

Reply:
@romain-grecourt Just to clarify, you mean to only have responses for the 500 status `ClassCastException` and `JsonParsingException` and the 400 status `JsonException`?
Thanks!

Reply:
I don't think we should check `JsonParsingException`, since it inherits `JsonException`. `ClassCastException` here is a side effect of the implementation for the JSONP reader, it does not seem right to have this code here ; instead the reader should be improved to catch the `ClassCastException` and re-throw something more specific (maybe a `JsonException`).

I.e, I suggest you only detect JsonException to return a 400 status, al other exceptions return a 500 status.
Use a generic message and do not include the exception message, e.g. `Invalid JSON` or `Internal error` and log the exception at a FINE level.

Feel free to create an issue to track the improvements to the JSONP reader for better handling JsonObject and JsonArray casting issues.

Reply:
@romain-grecourt Thanks! I updated with your requested changes and will look into filing that issue for the JSONP reader.

Reply:
That looks good to me.